scripts/firmware.py

```diff
@@ -0,0 +1,2144 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import subprocess
+import argparse
+import shutil
+import tempfile
+import fnmatch
+import platform
+import time
+
+from enum import Enum
+from time import sleep
+from typing import (
+    Optional,
+    TypedDict,
+    Callable,
+    NamedTuple,
+    Dict,
+    List,
+    Union,
+    Tuple,
+    Set,
+)
+
+HOME_PATH = os.path.expanduser("~")
+CONFIG_DIR: str = os.path.expanduser("~/printer_data/config")
+KLIPPY_LOG: str = os.path.expanduser("~/printer_data/logs/klippy.log")
+KLIPPER_DIR: str = os.path.expanduser("~/klipper")
+KATAPULT_DIR: str = os.path.expanduser("~/katapult")
+
+FLASHER_VERSION: str = "0.0.1"
+
+PAGE_WIDTH: int = 89  # Default global width
+
+is_advanced: bool = False
+
+
+class Color(Enum):
+    RESET = "\033[0m"
+    RED = "\033[91m"
+    GREEN = "\033[92m"
+    YELLOW = "\033[93m"
+    MAGENTA = "\033[95m"
+    CYAN = "\033[96m"
+
+
+class FlashMethod(str, Enum):
+    CAN = "CAN"
+    USB = "USB"
+    DFU = "DFU"
+
+
+# Define a custom namespace class
+class FirmwareNamespace(argparse.Namespace):
+    branch: str = "master"
+    debug: bool = False
+    type: bool = False
+    high_temp: bool = False
+    all: bool = False
+    kseries: bool = False
+    device: Optional[str] = None
+    flash: Optional[str] = None
+
+
+class Version(TypedDict):
```

Comment:
I don't think this is used anywhere?

Reply:
Version? or flash?

Reply:
When i remove the flash line i get type errors elsewhere

Reply:
Version, woops.